nakedret.go

```diff
@@ -42,129 +48,97 @@ func main() {
+	return err == nil
+}
+
+func checkNakedReturns(maxLength uint, shouldFix bool, fset *token.FileSet, files map[string]*ast.File) error {
+	hasNaked := false
 	retVis := &returnsVisitor{
 		f:         fset,
-		maxLength: *maxLength,
+		maxLength: maxLength,
+		shouldFix: shouldFix,
+		hasNaked:  &hasNaked,
 	}
 
+	updatedFiles := []*ast.File{}
+
 	for _, f := range files {
 		ast.Walk(retVis, f)
+		if hasNaked && shouldFix {
+			updatedFiles = append(updatedFiles, f)
+			hasNaked = false
+		}
 	}
 
-	return nil
-}
-
-func parseInput(args []string, fset *token.FileSet) ([]*ast.File, error) {
-	var directoryList []string
-	var fileMode bool
-	files := make([]*ast.File, 0)
-
-	if len(args) == 0 {
-		directoryList = append(directoryList, pwd)
-	} else {
-		for _, arg := range args {
-			if strings.HasSuffix(arg, "/...") && isDir(arg[:len(arg)-len("/...")]) {
-
-				for _, dirname := range allPackagesInFS(arg) {
-					directoryList = append(directoryList, dirname)
-				}
-
-			} else if isDir(arg) {
-				directoryList = append(directoryList, arg)
-
-			} else if exists(arg) {
-				if strings.HasSuffix(arg, ".go") {
-					fileMode = true
-					f, err := parser.ParseFile(fset, arg, nil, 0)
-					if err != nil {
-						return nil, err
-					}
-					files = append(files, f)
-				} else {
-					return nil, fmt.Errorf("invalid file %v specified", arg)
-				}
-			} else {
+	if shouldFix {
+		for _, f := range updatedFiles {
+			file := fset.File(f.Package)
+			reportFile := file.Name()
 
-				//TODO clean this up a bit
-				imPaths := importPaths([]string{arg})
-				for _, importPath := range imPaths {
-					pkg, err := build.Import(importPath, ".", 0)
-					if err != nil {
-						return nil, err
-					}
-					var stringFiles []string
-					stringFiles = append(stringFiles, pkg.GoFiles...)
-					// files = append(files, pkg.CgoFiles...)
-					stringFiles = append(stringFiles, pkg.TestGoFiles...)
-					if pkg.Dir != "." {
-						for i, f := range stringFiles {
-							stringFiles[i] = filepath.Join(pkg.Dir, f)
-						}
-					}
-
-					fileMode = true
-					for _, stringFile := range stringFiles {
-						f, err := parser.ParseFile(fset, stringFile, nil, 0)
-						if err != nil {
-							return nil, err
-						}
-						files = append(files, f)
-					}
-
-				}
-			}
-		}
-	}
+			fmt.Println(reportFile)
+			b := &bytes.Buffer{}
+			printer.Fprint(b, fset, f)
 
-	// if we're not in file mode, then we need to grab each and every package in each directory
-	// we can to grab all the files
-	if !fileMode {
-		for _, fpath := range directoryList {
-			pkgs, err := parser.ParseDir(fset, fpath, nil, 0)
+			formatted, err := format.Source(b.Bytes())
 			if err != nil {
-				return nil, err
+				formatted = b.Bytes()
+				log.Printf("got error while formatting: %v\n", err)
 			}
 
-			for _, pkg := range pkgs {
-				for _, f := range pkg.Files {
-					files = append(files, f)
-				}
-			}
+			ioutil.WriteFile(reportFile, formatted, os.ModeDevice)
```

Comment:
Why do we need `ModeDevice` for this write?

Reply:
I was following some example; but I can't seem to find it. I'll use `0644` as in the go by example: https://gobyexample.com/writing-files